src/databricks/sql/auth/oauth.py

```diff
@@ -38,12 +40,13 @@ def __get_redirect_url(redirect_port: int):
     def __fetch_well_known_config(idp_url: str):
         known_config_url = f"{idp_url}/.well-known/oauth-authorization-server"
         try:
-            response = requests.get(url=known_config_url)
+            response = requests.get(url=known_config_url, timeout=10)
```

Comment:
Generally we like to keep these timeouts configurable. Perhaps when instantiating the OAuthManager?